orttraining/orttraining/python/training/onnxblock/engine/training_module.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+
+from onnxruntime.capi import _pybind_state as C
+from onnxruntime.capi.onnxruntime_pybind11_state import OrtValueVector
+
+
+class TrainingModule:
+    """
+    Class for running Training.
+    This class is a wrapper of Module Class.
+    """
+
+    training: bool
+
+    def __init__(
+        self, train_model_uri, ckpt_uri, eval_model_uri=None, session_options=None, providers=None, **kwargs
+    ) -> None:
+        """
+        Initializes Model for Training.
+        __init__ will call an internatl function to create the model.
+        """
+        # TODO : Add support for bytes on train_model_uri and eval_model_uri.
+        self.training = True
+        self.fetches = OrtValueVector()
+        self._train_model_uri = train_model_uri
+        self._ckpt_uri = ckpt_uri
+        self._eval_model_uri = eval_model_uri
+        self._session_options = session_options
+        if providers is None:
+            self._providers = C.get_available_providers()
+
+        self._create_training_module()
+
+    def __call__(self, *input):
+
+        if self.training:
+            self._model.train_step(*input, self.fetches)
+        elif self.training == False:
+            self._model.eval_step(*input, self.fetches)
+
+        return self.fetches[0].numpy()
```

Comment:
What is the idea behind having fetches as a member variable?

Reply:
So users do not have to create an empty ortVector and pass it, so we just initialize it in the object creation time.